autoload/sj/go.vim

```diff
@@ -73,6 +126,30 @@ function! sj#go#JoinStruct()
     call add(arguments, argument)
   endfor
 
-  call sj#ReplaceMotion('va{', '{'.join(arguments, ', ').'}')
+  call sj#ReplaceMotion('va'.a:openBrace, a:openBrace . join(arguments, ', ') . a:closeBrace)
   return 1
 endfunction
+
+" Find start and end positions of braces which contains argument list.
+" This handles nested braces introduced by func types (see go_spec.rb).
+function! s:locateFuncBracesOnLine(line)
```

Comment:
Wouldn't `sj#LocateBracesOnLine('(', ')', ['goString', 'goComment'])` work well enough instead of this? It also handles nested brackets, so I think it should do the trick.